File: csharp/utils/NetworkData.cs
using System;
using System.Net;

namespace DesignAIRobotics.Comm.utils {
    class NetworkData {
        private static bool bigEndian = !BitConverter.IsLittleEndian;
        public static ulong shortSize = sizeof(short);
        public static ulong intSize = sizeof(int);
        public static ulong longSize = sizeof(long);
        public static ulong floatSize = sizeof(float);
        public static ulong doubleSize = sizeof(double);

        static public bool isBigEndian() {
            return bigEndian;
        }

        static public bool isLittleEndian() {
            return !isBigEndian();
        }

        public static void shortToNetworkBytes(byte[] buffer, int start, short value) {
            Utils.memcpy(buffer, (ulong) start, BitConverter.GetBytes(IPAddress.HostToNetworkOrder(value)), 0, shortSize);
        }

        public static short networkBytesToShort(byte[] buffer, int start) {
            return IPAddress.NetworkToHostOrder(BitConverter.ToInt16(buffer, start));
        }

        public static void intToNetworkBytes(byte[] buffer, int start, int value) {
            Utils.memcpy(buffer, (ulong) start, BitConverter.GetBytes(IPAddress.HostToNetworkOrder(value)), 0, intSize);
        }

        public static int networkBytesToInt(byte[] buffer, int start) {
            return IPAddress.NetworkToHostOrder(BitConverter.ToInt32(buffer, start));
        }

        public static void longLongToNetworkBytes(byte[] buffer, int start, long value) {
            Utils.memcpy(buffer, (ulong) start, BitConverter.GetBytes(IPAddress.HostToNetworkOrder(value)), 0, longSize);
        }

        public static long networkBytesToLongLong(byte[] buffer, int start) {
            return IPAddress.NetworkToHostOrder(BitConverter.ToInt64(buffer, start));
        }

        public static void floatToNetworkBytes(byte[] buffer, int start, double value) {
            Utils.memcpy(buffer, (ulong) start, BitConverter.GetBytes(value), 0, floatSize);
        }

        public static float networkBytesToFloat(byte[] buffer, int start) {
            return BitConverter.ToSingle(buffer, start);
        }

        public static void doubleToNetworkBytes(byte[] buffer, int start, double value) {
            Utils.memcpy(buffer, (ulong) start, BitConverter.GetBytes(value), 0, doubleSize);
        }

        public static double networkBytesToDouble(byte[] buffer, int start) {
            return BitConverter.ToDouble(buffer, start);
        }
    }
}

File: python/comm/comm_utils/NetworkData.py
import struct
from comm.comm_utils.utils import memcpy


class NetworkData:
    charSize = 1
    shortSize = 2
    intSize = 4
    longLongSize = 8
    floatSize = 4
    doubleSize = 8

    @staticmethod
    def charToNetworkBytes(buffer: bytes, start: int, value: int) -> bytes:
        return buffer[:start] + value.to_bytes(NetworkData.charSize, 'big') + buffer[start + NetworkData.charSize:]

    @staticmethod
    def networkBytesToChar(buffer: bytes, start: int) -> int:
        return int.from_bytes(buffer[start:start + NetworkData.charSize], 'big')

    @staticmethod
    def shortToNetworkBytes(buffer: bytes, start: int, value: int) -> bytes:
        return memcpy(buffer, start, value.to_bytes(NetworkData.shortSize, 'big'), 0, NetworkData.shortSize)

    @staticmethod
    def networkBytesToShort(buffer: bytes, start: int) -> int:
        return int.from_bytes(buffer[start:start + NetworkData.shortSize], 'big')

    @staticmethod
    def intToNetworkBytes(buffer: bytes, start: int, value: int) -> bytes:
        return memcpy(buffer, start, value.to_bytes(NetworkData.intSize, 'big'), 0, NetworkData.intSize)

    @staticmethod
    def networkBytesToInt(buffer: bytes, start: int) -> int:
        return int.from_bytes(buffer[start:start + NetworkData.intSize], 'big')

    @staticmethod
    def longLongToNetworkBytes(buffer: bytes, start: int, value: int) -> bytes:
        return memcpy(buffer, start, value.to_bytes(NetworkData.longLongSize, 'big'), 0, NetworkData.longLongSize)

    @staticmethod
    def networkBytesToLongLong(buffer: bytes, start: int) -> int:
        return int.from_bytes(buffer[start:start + NetworkData.longLongSize], 'big')

    @staticmethod
    def floatToNetworkBytes(buffer: bytes, start: int, value: float) -> bytes:
        return memcpy(buffer, start, struct.pack("<f", value), 0, NetworkData.floatSize)

    @staticmethod
    def networkBytesToFloat(buffer: bytes, start: int) -> float:
        return struct.unpack("<f", buffer[start:start + NetworkData.floatSize])[0]

    @staticmethod
    def doubleToNetworkBytes(buffer: bytes, start: int, value: float) -> bytes:
        return memcpy(buffer, start, struct.pack("<d", value), 0, NetworkData.doubleSize)

    @staticmethod
    def networkBytesToDouble(buffer: bytes, start: int) -> float:
        return struct.unpack("<d", buffer[start:start + NetworkData.doubleSize])[0]
